src/query/block/lazy.go

```diff
@@ -123,13 +130,13 @@ func (it *lazyStepIter) SeriesMeta() []SeriesMeta { return it.it.SeriesMeta() }
 func (it *lazyStepIter) Next() bool               { return it.it.Next() }
 
 func (it *lazyStepIter) Meta() Metadata {
-	return it.opts.MetaTransform(it.it.Meta())
+	return it.opts.MetaTransform()(it.it.Meta())
```

Comment:
This looks super gross haha, don't think there's anything we can do about it though

Reply:
Yeah can you make this look slightly better where every time you call it you do:
```go
meta := it.opts.MetaTransform()
return meta(it.it.Meta())
```